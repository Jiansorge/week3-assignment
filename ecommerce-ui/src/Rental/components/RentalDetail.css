.rental-detail {
    display: flex;
    flex-flow: row wrap;
    text-transform: uppercase;
    color: gray;
    font-size: 0.7rem;
    padding: 8px 0 4px;
    font-weight: bold;
}