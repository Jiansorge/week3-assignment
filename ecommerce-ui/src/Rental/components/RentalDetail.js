//  - housetype
//  - location city

import React from 'react';
import './RentalDetail.css'
import PropTypes from 'prop-types';

class RentalDetail extends React.Component {
  static propTypes = {
    rental: PropTypes.object.isRequired,
  }

    render() {

      return (
        <div className = "rental-detail">
          <p className="house-type">
            {this.props.rental.houseType}
            &nbsp; &#8226; &nbsp; 
            {this.props.rental.location.city}
          </p>
        </div>
      );
    }

  }
  export default RentalDetail;