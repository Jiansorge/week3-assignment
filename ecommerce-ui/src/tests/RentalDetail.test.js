import React from 'react';
import RentalDetail from '../Rental/components/RentalDetail.js';
import {shallow} from 'enzyme';
import airbnbs from '../airbnbs.json';


//import testData from './testData.json';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<Image />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('should match the snapshot', () => {
  const wrapper = shallow(<RentalDetail rental={airbnbs[0]} />);
  expect(wrapper).toMatchSnapshot();
});