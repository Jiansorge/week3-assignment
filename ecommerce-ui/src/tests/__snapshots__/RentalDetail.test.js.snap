// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="rental-detail"
>
  <p
    className="house-type"
  >
    Apartment
      •  
    New York
  </p>
</div>
`;
